### ISSUE TITLE
Undefined values in object are not handled

### ISSUE THREAD

**Issue by alerighi** [j-brooke/FracturedJsonJs#5]
By trying to serialize an object that contains `undefined` values the library will break:

```js
const formatter = new Formatter()
formatter.serialize({ a: undefined })
```

will produce the following exception:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'match')
    at stringToArray (/.../node_modules/eastasianwidth/eastasianwidth.js:273:17)
    at Object.eaw.length (/.../node_modules/eastasianwidth/eastasianwidth.js:277:20)
    at Formatter.StringWidthWithEastAsian [as _stringWidthFunc] (/.../node_modules/fracturedjsonjs/dist/fracturedjson.js:235:20)
    at Formatter.formatSimple (/.../node_modules/fracturedjsonjs/dist/fracturedjson.js:284:39)
    at Formatter.formatElement (/.../node_modules/fracturedjsonjs/dist/fracturedjson.js:271:34)
    at Formatter.formatObject (/.../node_modules/fracturedjsonjs/dist/fracturedjson.js:342:31)
    at Formatter.formatElement (/.../node_modules/fracturedjsonjs/dist/fracturedjson.js:269:34)
    at Formatter.serialize (/.../node_modules/fracturedjsonjs/dist/fracturedjson.js:221:42)
```

For now the workaround that I use is to do:

```js
formatter.serialize(JSON.parse(JSON.stringify({ a: undefined })))
```
To remove `undefined` values from an object.

**Comment by j-brooke** [comment]
Good catch - thanks.

FracturedJson should probably model its `undefined` handling after `JSON.stringify`.  How it behaves depends on the situation, it seems:

```
> JSON.stringify(null);
'null'
> JSON.stringify(undefined);
undefined
> JSON.stringify([null]);
'[null]'
> JSON.stringify([undefined]);
'[null]'
> JSON.stringify({a:null});
'{"a":null}'
> JSON.stringify({a:undefined});
'{}'
> JSON.stringify({null:null});
'{"null":null}'
> JSON.stringify({undefined:null});
'{"undefined":null}'
```

**Comment by j-brooke** [comment]
Fixed in version 2.2.1.
